go/interpreter/lfvm/instructions.go

```diff
@@ -862,27 +862,26 @@ func opExtcodehash(c *context) {
 	}
 }
 
-func checkInitCodeSize(c *context, size *uint256.Int) bool {
+// checkInitCodeSize checks the size of the init code.
+// An error is returned if size is greater than MaxInitCodeSize, or
+// if not enough gas is available.
+// Caller should handle error return and stop the execution.
+func checkInitCodeSize(c *context, size *uint256.Int) error {
```

Comment:
For a follow-up: Since the function is charging gas it should not just be called "check".

This function actually does a lot of stuff that should be divided up:
- check for maximum length depending on the revision
- if the size is larger than the maximum limit it consumes all the remaining gas (which it should probably not do; using all remaining gas in an error case should be handled in some central place)
- it charges gas for the init code size

This function looks to me like a prime candidate for being inlined into its call-side which after #781 should only be a single place. Wdyt?

Reply:
gas usage can definitely be inlined in genericCreate, but I like having the `MaxInitCodeSize` check  here. As for consuming all gas, I think it is an exceptional handling. But with proper error propagation it should easily be handled in `genericCreate` instead of here.  But yes, I like the little refactor